hazelcast/src/main/java/com/hazelcast/jet/impl/execution/init/ExecutionPlanBuilder.java

```diff
@@ -102,7 +102,11 @@ public static CompletableFuture<Map<MemberInfo, ExecutionPlan>> createExecutionP
                     .orElseThrow();
             partitionsByMember.computeIfAbsent(localMemberInfo, (i) -> {
                 nodeEngine.getLogger(ExecutionPlanBuilder.class).info("Adding coordinator to partition-pruned job members");
-                return new int[]{};
+                return nodeEngine.getPartitionService()
+                        .getMemberPartitionsIfAssigned(localMemberAddress)
```

Comment:
jet internally has mapping for all partitions in senders/receivers, because partition ids are used not only with IMap keys, but also with arbitrary keys.

what if partition is not assigned (IMap does not contain data?) but `findLocalPartitioningKey` returned such partition?

Reply:
AFAIK : no data -> no data flow on processors -> nothing happens

Reply:
what do you mean "no data"? Data in IMaps does not matter for that, flow is directed by `allToOne` and key specified in it.

Reply:
waiting for response

Reply:
> what if partition is not assigned (IMap does not contain data?) but findLocalPartitioningKey returned such partition?

I will answer frankly: I don't know. I created a test for that (IMap does not contain data), and it doesn't break. Currently I have no idea how to reproduce it in any other way.

Reply:
https://github.com/hazelcast/hazelcast/pull/24813/commits/96b0110c977e907bf2d85495319989801588cfd2

Reply:
> I will answer frankly: I don't know.

I also do not know (if I knew, I would not ask). Did you analyse the code? May be a bit time consuming (that is why I did not do that), but IMO worth of it, because otherwise we risk having intermittent and hard to reproduce errors.

Reply:
If we will have a problem with that, we definitely will have a corresponding exception. I analyzed the code, but couldn't find anything that made me wary about this solution. If you want to push & prioritize this case, I can do research and some tests to break it as a separate effort.

Reply:
after the changes we no longer use `getMemberPartitionsIfAssigned` so that possible problem is no more relevant